pkg/apis/pipeline/v1/container_validation.go

```diff
@@ -91,6 +91,9 @@ func (ref *Ref) Validate(ctx context.Context) (errs *apis.FieldError) {
 
 // RefNameLikeUrl checks if the name is url parsable and returns an error if it isn't.
 func RefNameLikeUrl(name string) error {
-	_, err := url.ParseRequestURI(name)
```

Comment:
This was a bit too strict where it didn't handle cases like:
`<scheme>://foo:v0`